test/framework/src/main/java/org/elasticsearch/test/rest/yaml/ClientYamlTestResponse.java

```diff
@@ -45,15 +45,21 @@
 
     private final Response response;
     private final byte[] body;
-    private final XContentType bodyContentType;
+    private XContentType bodyContentType;
     private ObjectPath parsedResponse;
     private String bodyAsString;
 
     public ClientYamlTestResponse(Response response) throws IOException {
         this.response = response;
         if (response.getEntity() != null) {
             String contentType = response.getHeader("Content-Type");
-            this.bodyContentType = XContentType.fromMediaType(contentType);
+            // todo pg it feels like there is a lot of 'ifs' around text, csv etc being returned. shouldn't we treat text as the same json and others?
+            // we are testing sql formats with this client, but don't have access to sql parsing code
+            try{
```

Comment:
this client does not have access to SQL code, but has to support these formats too.

Reply:
we are not throwing an exception when parsing in this PR